src/js/indexes/components/Files.js

```diff
@@ -0,0 +1,33 @@
+import { map } from "lodash-es";
+import React, { useMemo } from "react";
+import { connect } from "react-redux";
+import { Badge, BoxGroup, BoxGroupHeader } from "../../base";
+import { File } from "./File";
+
+export function Files({ files }) {
+    const fileComponents = useMemo(
```

Comment:
Use of `useMemo` is not advisable here.

1. Unnecessary re-render will already be prevented by `react-redux`. No render occurs unless the values returned from `mapStateToProps` change.
2. Conventional React wisdom: optimize when required, not by deefault. There will be little benefit to `useMemo` here even without `react-redux`. The added complexity and memory usage is not worth it.

I didn't know `useMemo` could return components. Good to know!

Reply:
Good points, changed.